### PR TITLE
Merge payment gateway suggestion classes into single source of truth

### LINKED ISSUE #8223: Payments recommendation card - introduce more extensions
**Description:** now that the payments recommendation placement seems pretty stable in terms of adoption, we want to include more payments partners. We want to start by replicating the same rules that are applied to the payments task.

**Acceptance criteria:**
- [ ] Replicate the same rules we have in place for the payments task. Including the order in which we should show each extension
- [ ] The only exception is WCPay, which should be kept in the other location (at the top of the payments table)

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #8223 

* Removes additional payment gateway REST APIs and consolidates into a single source of truth.
* Updates the frontend to use the current REST API

### Screenshots

Screen Shot 2022-03-02 at 11 50 48 AM

### Detailed test instructions:

**Make use of updated wc.com changes**
*while testing this with this branch, it would be appreciated if you check and approve the wc.com PR as well.
1. Load this woocommerce.com branch locally - 12649-gh-Automattic/woocommerce.com
2. Make sure this transient is deleted `woocommerce_admin_payment_gateway_suggestions_specs`
3. Add this filter to your mu-plugin to point to your local woocommerce.com environment:
```
function change_datasource_to_local_wc( $data_sources ) {
	$new_data_sources = array();
	foreach ( $data_sources as $url ) {
		if ( strpos($url, 'woocommerce.com') !== false ) {
			$new_domain = 'http://woocommerce.test';
			$new_url = preg_replace('/^(.*woocommerce.com)/', $new_domain, $url);
			$new_data_sources[] = $new_url;
		} else {
			$new_data_sources[] = $url;
		}
	}
	return $new_data_sources;
}
add_filter( 'data_source_poller_data_sources', 'change_datasource_to_local_wc', 10 );
```

**Ready to start testing now**
1. Enable marketplace suggestions `wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
5. Visit the payments settings screen - `wp-admin/admin.php?page=wc-settings&tab=checkout`
6. Make sure that payment suggestions are loaded beneath the installed payment providers
7. Switch around to various countries to check that the relevant gateways are shown
8. Make sure gateways without plugins (COD & BACS) are not shown in the list
9. Install a gateway
10. Verify that you are redirected to that payment gateway's settings screen after install
11. Navigate back to the payment suggestions list
12. Check that the installed gateway is no longer shown in the suggestions list

### CODE REVIEW

File: client/payments/payment-recommendations.scss
Comment (louwie17) [inline client/payments/payment-recommendations.scss:65]: Could we also center the images vertically, there seems to be quite a bit of space below some of them:
Screen Shot 2022-02-07 at 1 18 43 PM

File: client/payments/payment-recommendations.tsx
Comment (louwie17) [inline client/payments/payment-recommendations.tsx:164]: Given the `woocommerce_payments` promotion is still listed as a feature, could we add this condition `window.wcAdminFeatures['wc-pay-promotion']`. Just incase we decide to disable it.

Something is to be said for when we move things out of the feature folder, but that is another discussion to be had :)

File: src/API/PaymentGatewaySuggestions.php
Comment (louwie17) [inline src/API/PaymentGatewaySuggestions.php:54]: This `'/payment-gateway-suggestions/dismiss'` should be `'/dismiss'` correct? Currently the dismiss action causes a 404 on the front-end.

**Review by louwie17 — COMMENTED** [review]
Overall nice work here @joshuatf, the consolidation is nice, and good to see there are more removed lines then added :)
I left several comments, everything did test well aside from the `dismiss` button, but that is a simple fix.

I know @pmcpinto mentioned to use the [same copy as the payments task](https://github.com/woocommerce/woocommerce-admin/issues/8223#issuecomment-1027791587), but does this also go for the current Stripe & Paypal options that were previously displayed? (I guess that question is for @pmcpinto )
Also the current endpoint uses different logo's, square one's in particular compared to the wordy one's from the payments task. In my opinion I think the square one's fit this box better, but we should decide on that as well?
cc @jarekmorawski in relation to the image logo's.

Here is the difference between the logos, I guess you can see the title and copy differences here as well:
Currently:
Screen Shot 2022-02-07 at 2 50 48 PM

New:
Screen Shot 2022-02-07 at 2 49 39 PM

File: client/payments/payment-recommendations.scss
Comment (joshuatf) [reply client/payments/payment-recommendations.scss:65]: Seems like a reasonable change.  Updated to the whole row aligning center (including the action button) to match the suggestions in the task list.

File: client/payments/payment-recommendations.tsx
Comment (joshuatf) [reply client/payments/payment-recommendations.tsx:119]: We previously were relying on the `setup-link` returned in the gateway response.  This is brittle since the settings page can be changed within the plugin.

The new check relies on fetching the gateway details after it has been installed and using its self-defined settings URL.

Resolvers in WP data (or at least the currently bundled version) do not return promises so we cannot await the payment gateway resolution.  I agree that it's slightly more confusing and I'm open to other pattern suggestions, but I believe this is the commonly used pattern.

On a side note, I think I recall someone mentioning that newer versions of WP data may allow us to await resolvers.

File: client/payments/payment-recommendations.tsx
Comment (joshuatf) [reply client/payments/payment-recommendations.tsx]: Nice!  Great idea.  Added.

File: src/API/PaymentGatewaySuggestions.php
Comment (joshuatf) [reply src/API/PaymentGatewaySuggestions.php]: Good catch.  Updated and also optimistically dismissed the recommendations client-side to avoid the delay we were seeing.

There is a lot more renaming that could be done around "recommendations" and moving methods within data stores, but I'm going to try and keep the changes in this PR to as little as possible and we can follow-up on those.

**Comment by joshuatf** [comment]
>I know @pmcpinto mentioned to use the https://github.com/woocommerce/woocommerce-admin/issues/8223#issuecomment-1027791587, but does this also go for the current Stripe & Paypal options that were previously displayed? (I guess that question is for @pmcpinto )

Good question.  @pmcpinto Also worth noting that I updated "Recommended ways to get paid" to "Additional ways to get paid."  No strong opinions on this, but was trying to discern between the one "recommended" gateway in the list and other suggestions and found this copy confusing.

**Comment by pmcpinto** [comment]
> I know @pmcpinto mentioned to use the https://github.com/woocommerce/woocommerce-admin/issues/8223#issuecomment-1027791587, but does this also go for the current Stripe & Paypal options that were previously displayed? (I guess that question is for @pmcpinto )

Yes. We should use the Stripe and PP task list copy should be applied on this card.

> Good question. @pmcpinto Also worth noting that I updated "Recommended ways to get paid" to "Additional ways to get paid." No strong opinions on this, but was trying to discern between the one "recommended" gateway in the list and other suggestions and found this copy confusing.

This copy update makes sense 👍 

> Also the current endpoint uses different logo's, square one's in particular compared to the wordy one's from the payments task. In my opinion I think the square one's fit this box better, but we should decide on that as well?
cc @jarekmorawski in relation to the image logo's.

The logos from the payments task look too distracting on this card. @jarekmorawski  what do you think?

**Review by louwie17 — COMMENTED** [review]
Thanks for the updates @joshuatf, things are looking great now aside from the logo's.
I do think those should be updated before we merge it, or at-least some feedback from @jarekmorawski , outside of that you got my approval.

**Comment by jarekmorawski** [comment]
> The logos from the payments task look too distracting on this card. @jarekmorawski what do you think?

I agree and I think that we should stick to rectangular icons that have a smaller footprint (like in the original design).

**Comment by joshuatf** [comment]
Thanks for the input, everyone!

I think thumbnails sound good as well.  We'll need to:
* Add thumbnails to our default gateways
* Add them to the WCCOM side
* Consume the thumbnails in these frontend components

**Comment by louwie17** [comment]
> I agree and I think that we should stick to rectangular icons that have a smaller footprint (like in the original design).

@jarekmorawski do you happen to have these rectangular icons for the different payment providers? Or is there a Figma I can get them from? Elizabeth had given them to me last time, but just for Stripe, Paypal, and WooCommerce Payments ([original comment](https://github.com/woocommerce/woocommerce-admin/issues/6684#issuecomment-814187870))

**Comment by jarekmorawski** [comment]
Hi @louwie17! Here's the Figma file with the icons: 7QOkvxzz1ylS3QBS4zqW8R-fi-8947%3A35475

**Comment by louwie17** [comment]
> Hi @louwie17! Here's the Figma file with the icons: [7QOkvxzz1ylS3QBS4zqW8R-fi-8947%3A35475](https://href.li/?https://www.figma.com/file/7QOkvxzz1ylS3QBS4zqW8R/?node-id=8947%3A35475)

Thanks @jarekmorawski, that one is handy to have, but it does only provide the Paypal, Stripe, and WooCommerce Payments icons. Is there a place to get the square icons for Payfast, Paystack, Klarna, Mollie, PayU, Razorpay, Eway? (you can find them [in this list](https://woocommerce.com/wp-json/wccom/payment-gateway-suggestions/1.0/suggestions.json)).
If we don't have these readily available, I can fallback to the current image (the same that we use in the payments task ) for now, and then we can update them as they become available (as this would be a change in wc.com and not relient on our releases).

**Comment by jarekmorawski** [comment]
> If we don't have these readily available, I can fallback to the current image (the same that we use in the payments task ) for now, and then we can update them as they become available (as this would be a change in wc.com and not relient on our releases).

Sounds good! I don't have the icons at hand and it's likely that parts of this page will change in the near future. Let's wait with the update until then.

**Review by jacob-sewell — APPROVED** [review]
Tested as described, looks good! Way to go!

**Pull request merged**
